### ISSUE TITLE
[Feature] Support for ESLint v10

### ISSUE THREAD

**Issue by NishargShah** [eslint-stylistic/eslint-stylistic#1124]
### Clear and concise description of the problem

Support ESLint v10 for `@stylistic/eslint-plugin`

### Suggested solution

NA

### Additional context

https://eslint.org/blog/2026/02/eslint-v10.0.0-released

### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [x] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by SalahAdDin** [comment]
It seems the new configuration system is failing on ESLint 10:
```shell
> eslint .


Oops! Something went wrong! :(

ESLint: 10.0.0

TypeError: Error while loading rule '@stylistic/comma-dangle': Cannot read properties of undefined (reading 'ecmaVersion')
Occurred while linting G:\Projects\sites\central-mosque-bogota\frontend\.oxfmtrc.json
    at create (file:///G:/Projects/sites/central-mosque-bogota/frontend/node_modules/.pnpm/@stylistic+eslint-plugin@5.8.0_eslint@10.0.0_jiti@2.6.1_/node_modules/@stylistic/eslint-plugin/dist/rules/comma-dangle.js:78:118)
    at Object.create (file:///G:/Projects/sites/central-mosque-bogota/frontend/node_modules/.pnpm/@stylistic+eslint-plugin@5.8.0_eslint@10.0.0_jiti@2.6.1_/node_modules/@stylistic/eslint-plugin/dist/utils.js:478:11)
    at createRuleListeners (G:\Projects\sites\central-mosque-bogota\frontend\node_modules\.pnpm\eslint@10.0.0_jiti@2.6.1\node_modules\eslint\lib\linter\linter.js:497:15)
    at G:\Projects\sites\central-mosque-bogota\frontend\node_modules\.pnpm\eslint@10.0.0_jiti@2.6.1\node_modules\eslint\lib\linter\linter.js:623:7
    at Array.forEach (<anonymous>)
    at runRules (G:\Projects\sites\central-mosque-bogota\frontend\node_modules\.pnpm\eslint@10.0.0_jiti@2.6.1\node_modules\eslint\lib\linter\linter.js:557:31)
    at #flatVerifyWithoutProcessors (G:\Projects\sites\central-mosque-bogota\frontend\node_modules\.pnpm\eslint@10.0.0_jiti@2.6.1\node_modules\eslint\lib\linter\linter.js:1248:4)
    at Linter._verifyWithFlatConfigArrayAndWithoutProcessors (G:\Projects\sites\central-mosque-bogota\frontend\node_modules\.pnpm\eslint@10.0.0_jiti@2.6.1\node_modules\eslint\lib\linter\linter.js:1333:43)
    at Linter._verifyWithFlatConfigArray (G:\Projects\sites\central-mosque-bogota\frontend\node_modules\.pnpm\eslint@10.0.0_jiti@2.6.1\node_modules\eslint\lib\linter\linter.js:1400:15)
    at Linter.verify (G:\Projects\sites\central-mosque-bogota\frontend\node_modules\.pnpm\eslint@10.0.0_jiti@2.6.1\node_modules\eslint\lib\linter\linter.js:861:9)
 ELIFECYCLE  Command failed with exit code 2.
```

For both ways to do it:
```ts
  stylistic.configs.customize({
    // the following options are the default values
    indent: 2,
    quotes: "double",
    semi: true,
    jsx: true,
    // ...
  }),

// or
  stylistic.configs.recommended,
```

Right not using it as a plugin does not give an error:
```ts
  {
plugins: { js, "import-x": importX, "@stylistic": stylistic },
    extends: [
      "js/recommended",
      "import-x/recommended",
      "@stylistic/recommended",
    ],
    rules: {
 "@stylistic/comma-dangle": [
        "error",
        {
          arrays: "only-multiline",
          objects: "only-multiline",
          imports: "only-multiline",
          exports: "only-multiline",
          functions: "never",
        },
      ],
      "@stylistic/function-paren-newline": "warn",
      "@stylistic/implicit-arrow-linebreak": "warn",
      "@stylistic/linebreak-style": ["error", "unix"],
      "@stylistic/operator-linebreak": "warn",
      "@stylistic/object-curly-newline": [
        "error",
        {
          ExportDeclaration: { multiline: true, minProperties: 5 },
        },
      ],
      "@stylistic/quotes": ["error", "double"],
      "@stylistic/semi": ["error", "always"],
    },
  },
```

**Comment by 9romise** [comment]
> Occurred while linting G:\Projects\sites\central-mosque-bogota\frontend\.oxfmtrc.json

@SalahAdDin I think it's not related to supporting ESLint 10.
The issue you faced is that you're using the `comma-dangle` rule on JSON files, which doesn't have official support for JSON.
If your configuration works in ESLint 9, could you provide a minimal reproduction?

Something may related: #1137

**Comment by SalahAdDin** [comment]
> > Occurred while linting G:\Projects\sites\central-mosque-bogota\frontend.oxfmtrc.json
> 
> [@SalahAdDin](https://github.com/SalahAdDin) I think it's not related to supporting ESLint 10. The issue you faced is that you're using the `comma-dangle` rule on JSON files, which doesn't have official support for JSON. If your configuration works in ESLint 9, could you provide a minimal reproduction?
> 
> Something may related: [#1137](https://github.com/eslint-stylistic/eslint-stylistic/issues/1137)

But I'm using that rule only for JS/TS scripts!
